scripts/mem_usage.py

```diff
@@ -0,0 +1,162 @@
+
+
+def print_pager_stat(name, size):
+    size_kib = size / 1024
+    if args.raw:
+        printf('%d ', size)
+    else:
+        printf('%-36s size %.8X %3d KiB\n', name, size, size_kib)
+
+
+def readelf_cmd():
+    return os.getenv('CROSS_COMPILE', '') + 'readelf'
+
+
+def main():
+    global args
+
+    in_shdr = False
+    sects = []
+    init_size = 0
+    paged_size = 0
+    unpaged_size = 0
+    unpaged_no_heap_size = 0
+
+    args = get_args()
+    env = os.environ.copy()
+    env['LC_ALL'] = 'C'
+    readelf = subprocess.Popen([readelf_cmd(), '-S', '-W', args.tee_elf],
+                               stdout=subprocess.PIPE, env=env,
+                               universal_newlines=True)
+    for line in iter(readelf.stdout.readline, ''):
+        if 'Section Headers:' in line:
+            in_shdr = True
+            continue
+        if 'Key to Flags:' in line:
+            in_shdr = False
+            continue
+        if in_shdr:
+            words = line.split()
+            if words[0] == '[':
+                words.pop(0)
+            try:
+                (_, name, _, addr, offs, size, _,
+                 flags) = words[:8]
+            except:
+                continue
+            if (flags == 'AX' or flags == 'WA' or flags == 'A' or
+                    flags == 'AL'):
+                sects.append({'name': name, 'addr': addr,
+                              'offs': offs, 'size': size})
+    for sect in sects:
+        if sect['addr'] != 0:
+            first_addr = sect['addr']
+            break
+    last_addr = sects[-1]['addr']
+    last_size = sects[-1]['size']
+
+    ram_usage = int(last_addr, 16) + int(last_size, 16) - int(first_addr, 16)
+    print_sect('RAM Usage', int(first_addr, 16), ram_usage, True, True)
+
+    last_addr = 0
+    last_size = 0
+    for sect in sects:
+        name = sect['name']
+        addr = int(sect['addr'], 16)
+        size = int(sect['size'], 16)
+
+        if last_addr != 0 and addr != last_addr + last_size:
+            print_sect('*hole*', last_addr + last_size,
+                       addr - (last_addr + last_size))
+        print_sect(name, addr, size)
+        if name.endswith('_init'):
+            init_size += size
+        elif name.endswith('_pageable'):
+            paged_size += size
+        else:
+            if not name.startswith('.heap'):
+                unpaged_no_heap_size += size
+            unpaged_size += size
+        last_addr = addr
+        last_size = size
+
+    if args.all or args.init:
+        print_pager_stat('Init sections (.*_init)', init_size)
+    if args.all or args.paged:
+        print_pager_stat('Paged sections (.*_pageable)', paged_size)
+    if args.all or args.unpaged:
+        print_pager_stat('Unpaged sections ', unpaged_size)
+    if args.all or args.unpaged_no_heap:
+        print_pager_stat('Unpaged sections (code + data only)',
+                         unpaged_no_heap_size)
+    if (args.raw and (args.all or args.init or args.paged or
```

Comment:
Could be good to show the pager pool size (I think we may need to produce a reference in the code to get it from the symbols in the ELF). Maybe in a later change.